src/main.cpp

```diff
@@ -5921,23 +5927,41 @@ bool ProcessMessage(CNode* pfrom, std::string strCommand, CDataStream& vRecv, in
     }
 
 
-    else if (strCommand == NetMsgType::XPEDITEDREQUEST)
+    else if (strCommand == NetMsgType::XPEDITEDREQUEST && IsThinBlocksEnabled())
```

Comment:
If conditions like (&& IsThinBlocksEnabled()) are added to these top level if statements, then the code will fall through to the final else that issues this log:
        // Ignore unknown commands for extensibility
        LogPrint("net", "Unknown command \"%s\" from peer=%d\n", SanitizeString(strCommand), pfrom->id);

I don't think we want that log printed if !IsThinBlocksEnabled().  Let's put the IsThinBlocksEnabled() in HandleExpeditedRequest()